### ISSUE TITLE
Optional isPresent()

### ISSUE THREAD

**Issue by sherrycqu** [nus-cs2030/2122-s2#110]
I used the .isPresent() method of Optional class but codecrunch says cannot find symbol. I wonder why is that so... is it we are not allowed to use this method..

**Comment by szeying02** [comment]
Yup! `isPresent()`, `isEmpty()` and `get()` aren't allowed.

**Comment by sherrycqu** [comment]
then, is there any better ways for us to check the presence of an object in an immutablemap object?

**Comment by szeying02** [comment]
Unfortunately, the point of `Optional` is so that you don't need to check the presence of the object. The best way around this is to consider the other methods in the Optional API. All the best!

**Comment by michael192002** [comment]
A useful method that you might find helpful is the OrElse(T t) method in the Optional API, which manipulates the values based on its presence. Hope it helps!

**Comment by Ohnojiso** [comment]
Ya orElse can help you achieve this. It is also mentioned in the task text as a hint, so I think maybe you can check the java API for orElse n read its description to see how you can incorporate it in your implementation.

**Comment by Darien2307** [comment]
`jshell> roster.get("Natasha").flatMap(x -> x.get("CS2040")).flatMap(x -> x.get("Lab1")).map(Assessment::getGrade)`
`$.. ==> Optional[B]`
Hi, believe this is abt Lab6. Noticed the qn gave the above. So your `getGrade(String student, String module,...)` method just has to do the exact i.e.
`return get(student).flatMap(x -> x.get(module)).flatMap...map(x -> x.getGrade()).orElse(String.format("No such record ",...))`

**Comment by sherrycqu** [comment]
@michael192002 @Ohnojiso @Darien2307  Yes, I have managed to solve the qn without isPresent().  flatmap() and .orElse() is definitely very helpful!! Thanks!
